generators/server/templates/src/test/java/package/config/TestContainersSpringContextCustomizerFactory.java.ejs

```diff
@@ -120,6 +123,23 @@ public class TestContainersSpringContextCustomizerFactory implements ContextCust
                 testValues = testValues.and("spring.cloud.stream.kafka.binder.brokers=" + kafkaBean.getKafkaContainer().getHost() + ':' + kafkaBean.getKafkaContainer().getMappedPort(KafkaContainer.KAFKA_PORT));
             }
 <%_ } _%>
+<%_ if (messageBrokerPulsar) { _%>
+            EmbeddedPulsar pulsarAnnotation = AnnotatedElementUtils.findMergedAnnotation(testClass, EmbeddedPulsar.class);
+            if (null != pulsarAnnotation) {
+                log.debug("detected the EmbeddedPulsar annotation on class {}", testClass.getName());
+                log.info("Warming up the Pulsar instance");
+                if (null == pulsarBean) {
+                    pulsarBean = beanFactory.createBean(PulsarTestContainer.class);
+                    beanFactory.registerSingleton(PulsarTestContainer.class.getName(), pulsarBean);
+                    // ((DefaultListableBeanFactory)beanFactory).registerDisposableBean(PulsarTestContainer.class.getName(), pulsarBean);
+                }
+                testValues = 
+                    testValues.and(
+                        "spring.pulsar.client.service-url=" + pulsarBean.getPulsarContainer().getPulsarBrokerUrl(),
+                        "spring.pulsar.administration.service-url=" + pulsarBean.getPulsarContainer().getHttpServiceUrl()
+                    );
+            }
+<%_ } _%>
```

Comment:
Extract to its own file like [RedisTestContainersSpringContextCustomizerFactory.java.ejs ](https://github.com/jhipster/generator-jhipster/blob/main/generators/spring-cache/templates/src/test/java/package/config/RedisTestContainersSpringContextCustomizerFactory.java.ejs).
And inject like

https://github.com/jhipster/generator-jhipster/blob/f3d8124aad13ce33cbc4e8d48f3fd4e8e46da235/generators/spring-cache/generator.mts#L56-L59

Reply:
done